### PR TITLE
Fix/mobl 880 in app enhancements

### PR DESCRIPTION
**Pull request by ketanshikharebsft**
Download and cache images for in-app and show
Download the html content, load it and show in-app
Added fix for slidein inapp bottom safearea view 
Show close button based on API response value else show it with default setting
Added icon radius to slide in icon
Added icon padding to slide in icon 
Added auto height - auto width support for modal background image
Pass push notification userinfo to the open url method
Rearranged the BlueshiftINAppNotificationManager methods
Added logs

### CODE REVIEW

File: BlueShift-iOS-SDK/InApps/BlueShiftNotificationModalViewController.m [hunk 1]
```diff
@@ -439,9 +439,71 @@ - (CGFloat)getActionButtonXPosition:(UIView *)parentView childWidth:(CGFloat)wid
     : xPadding;
 }
 
+- (CGSize)getAutoImageSizeForNotificationView {
+    float width = 0;
+    float height = 0;
+    
+    // Check if this modal is image modal
+    if ([self isBackgroundImagePresentForNotification:self.notification] && (self.notification.templateStyle.width < 0 || self.notification.templateStyle.height < 0)) {
```
Comment (rahulrvp) [inline BlueShift-iOS-SDK/InApps/BlueShiftNotificationModalViewController.m:447]: this means if both width and height are a positive number, we will not respect the image's aspect ratio. right?

Comment (rahulrvp) [reply BlueShift-iOS-SDK/InApps/BlueShiftNotificationModalViewController.m:447]: in android, I am treating the positive height and width as my maximum space available. ex; if the width is 80% I will not take more than 80%. But width can be less than 80% based on the image's aspect ratio.

Comment (ketanshikharebsft) [reply BlueShift-iOS-SDK/InApps/BlueShiftNotificationModalViewController.m:447]: iOS implementation is little different, if the width height are in +ve then those values will be used irrespective of aspect ratio.

**Review by rahulrvp — APPROVED** [review]
LGTM

**Pull request merged**